op-challenger/game/keccak/matrix/matrix.go

```diff
@@ -53,15 +53,15 @@ func (d *StateMatrix) AbsorbNextLeaf(in io.Reader) error {
 			final = true
 			break
 		} else if err != nil {
-			return err
+			return nil, err
 		}
 		read += n
 	}
 	d.AbsorbLeaf(data[:read], final)
 	if final {
-		return io.EOF
+		return data[:read], io.EOF
 	}
-	return nil
+	return data[:LeafSize], nil
```

Comment:
I'd just store `data[:read]` as a var and return that so its always in sync:

```
    leafData := data[:read]
    d.AbsorbLeaf(leafData, final)
 	if final {
 		return leafData, io.EOF
 	}
 	return leafData, nil
```

Reply:
Good callout, will do!